File: spring-security-token/src/main/java/com/chen/demo/Controller/UserController.java
package com.chen.demo.Controller;

import com.chen.demo.entity.User;
import org.springframework.security.access.prepost.PreAuthorize;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
public class UserController {

    @GetMapping("/user-error")
    @PreAuthorize("hasAuthority('readss')")
    public User getUserError() {
        User user = new User();
        user.setName("error User");
        user.setGender(new Byte("1"));
        user.setEmail("123456@qq.com");

        return user;
    }

    @GetMapping("/user-authorize")
    @PreAuthorize("hasAuthority('read')")
    public User getUserAuthorize() {
        User user = new User();
        user.setName("user-authorize");
        user.setGender(new Byte("1"));
        user.setEmail("123456@qq.com");

        return user;
    }

    @GetMapping("/user")
    public User getUser() {
        User user = new User();
        user.setName("Defalut User");
        user.setGender(new Byte("1"));
        user.setEmail("123456@qq.com");

        return user;
    }


}

File: sprng-jcahce/src/main/kotlin/com/chen/demo/sprngjcahce/controller/UserController.kt
package com.chen.demo.sprngjcahce.controller

import com.chen.demo.sprngjcahce.entity.User
import com.chen.demo.sprngjcahce.service.UserService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.Mapping
import org.springframework.web.bind.annotation.RestController

@RestController
class UserController {
    @Autowired
    lateinit var userService: UserService

    @GetMapping("/")
    fun getUser(): ResponseEntity<User> {
        val user = userService.getUser()
        return ResponseEntity.ok(user)
    }
}